src/jit/gentree.h

```diff
@@ -5099,23 +5099,22 @@ inline GenTreePtr GenTree::gtGetOp2()
 
 inline GenTreePtr GenTree::gtEffectiveVal(bool commaOnly)
 {
-    switch (gtOper)
+    GenTree* effectiveVal = this;
+    for (;;)
     {
-        case GT_COMMA:
-            return gtOp.gtOp2->gtEffectiveVal(commaOnly);
-
-        case GT_NOP:
-            if (!commaOnly && gtOp.gtOp1 != nullptr)
-            {
-                return gtOp.gtOp1->gtEffectiveVal();
-            }
-            break;
-
-        default:
-            break;
+        if (effectiveVal->gtOper == GT_COMMA)
+        {
+            effectiveVal = effectiveVal->gtOp.gtOp2;
```

Comment:
Say we have the following tree node
treeNode = GT_COMMA1(op1A, op2=GT_COMMA2(Op1B, op2=GT_COMMA3(op1, op2)))

That is three GT_COMMA nodes nested.

Earlier treeNode->gtEffectiveVal() would return op2

Now with this change treeNode->gtEffectiveVal() would return GT_COMMA2 and caller would have to iterate to reach op2.  

But I see this method is called from many places within Jit codebase with the assumption that they will get op2 instead of GT_COMMA2.  Won't this change break those callers?